File: app/src/main/java/com/example/network_termproject/recycler/SelectAdapter.java
package com.example.network_termproject.recycler;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.network_termproject.databinding.SelectFriendItemLayoutBinding;

import java.util.ArrayList;

public class SelectAdapter extends RecyclerView.Adapter<SelectAdapter.SelectViewHolder> {

    private ArrayList<String> users;
    private ArrayList<SelectViewHolder> views;

    public SelectAdapter(ArrayList<String> users) {
        this.users = users;
        views = new ArrayList<>();
    }

    @NonNull
    @Override
    public SelectViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        SelectViewHolder viewHolder = new SelectViewHolder(SelectFriendItemLayoutBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
        views.add(viewHolder);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull SelectViewHolder holder, int position) {
        holder.bind(users.get(position));
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    public ArrayList<String> getCheckedUsers() {
        ArrayList<String> checkedUsers = new ArrayList<>();
        for (SelectViewHolder viewHolder : views) {
            if (viewHolder.binding.selectCheckBox.isChecked()) {
                checkedUsers.add(viewHolder.binding.selectNameTextView.getText().toString());
            }
        }
        return checkedUsers;
    }

    class SelectViewHolder extends RecyclerView.ViewHolder {

        private SelectFriendItemLayoutBinding binding;

        public SelectViewHolder(SelectFriendItemLayoutBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(String name) {
            binding.selectNameTextView.setText(name);
            binding.selectCheckBox.setChecked(false);
        }
    }
}

File: app/src/main/java/com/example/network_termproject/recycler/SelectAdapter.kt
package com.example.network_termproject.recycler

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.network_termproject.databinding.SelectFriendItemLayoutBinding
import com.example.network_termproject.recycler.SelectAdapter.SelectViewHolder
import java.util.*

class SelectAdapter(private val users: ArrayList<String>) : RecyclerView.Adapter<SelectViewHolder>() {

    private var views: ArrayList<SelectViewHolder>? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SelectViewHolder {
        val viewHolder = SelectViewHolder(SelectFriendItemLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false))
        views!!.add(viewHolder)
        return viewHolder
    }

    override fun onBindViewHolder(holder: SelectViewHolder, position: Int) {
        holder.bind(users[position])
    }

    override fun getItemCount(): Int {
        return users.size
    }

    val checkedUsers: ArrayList<String>
        get() {
            val checkedUsers = ArrayList<String>()
            for (viewHolder in views!!) {
                if (viewHolder.binding.selectCheckBox.isChecked) {
                    checkedUsers.add(viewHolder.binding.selectNameTextView.text.toString())
                }
            }
            return checkedUsers
        }

    class SelectViewHolder(val binding: SelectFriendItemLayoutBinding) : ViewHolder(binding.root) {
        fun bind(name: String?) {
            binding.selectNameTextView.text = name
            binding.selectCheckBox.isChecked = false
        }
    }

    init {
        views = ArrayList()
    }
}